code401challenges/src/main/java/code401challenges/linkedlist/LinkedList.java

```diff
@@ -99,20 +99,20 @@ public void insertafter(T newValue, T curValue) {
 
     //method for the Linked List class that takes a number, k, as a parameter.
     // Return the node’s value that is k from the end of the linked list
-    public void kthFromEnd(int k) {
+    public T kthFromEnd(int k) {
         Node<T> mainPtr = this.head;
         Node<T> current = this.head;
         int count = 0;
 
-        if(k <= 0){
-            System.out.println("k is invalid number");
+        if(k <= 0 || k < 0){
```

Comment:
If `k` is `0` then this method should return the last item in the linked list. This code will throw an error if the user passes in `0`.